packages/repository/test/unit/repositories/atomic.repository.unit.ts

```diff
@@ -0,0 +1,351 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  Entity,
+  EntityNotFoundError,
+  juggler,
+  ModelDefinition,
+  DefaultAtomicCrudRepository,
+} from '../../..';
+
+describe('AtomicCrudRepository', () => {
+  let ds: juggler.DataSource;
+
+  class Note extends Entity {
+    static definition = new ModelDefinition({
+      name: 'Note',
+      properties: {
+        title: 'string',
+        content: 'string',
+        id: {name: 'id', type: 'number', id: true},
+      },
+    });
+
+    title?: string;
+    content?: string;
+    id: number;
+
+    constructor(data: Partial<Note>) {
+      super(data);
+    }
+  }
+
+  beforeEach(() => {
+    ds = new juggler.DataSource({
+      name: 'db',
+      connector: 'memory',
+    });
+  });
+
+  context('constructor', () => {
+    class ShoppingList extends Entity {
+      static definition = new ModelDefinition({
+        name: 'ShoppingList',
+        properties: {
+          id: {
+            type: 'number',
+            id: true,
+          },
+          created: {
+            type: () => Date,
+          },
+          toBuy: {
+            type: 'array',
+            itemType: 'string',
+          },
+          toVisit: {
+            type: Array,
+            itemType: () => String,
+          },
+        },
+      });
+
+      created: Date;
+      toBuy: String[];
+      toVisit: String[];
+    }
+
+    it('converts PropertyDefinition with array type', () => {
```

Comment:
I copied the tests from `DefaultCrudRepository` unit test file because I was worried code coverage would drop if I didn't have them and since we extend from it.

Reply:
Well, as far as I understand code coverage, it counts lines of source code. As long as there is a test executing the source code behind `DefaultCrudRepository` in the file `legacy-juggler-bridge.ts`, it does not matter whether this test creates an instance of `DefaultCrudRepository` directly or an instance of a class inheriting from this Repository implementation.

Reply:
Ah yeah that is correct, I misunderstood how it worked. Thanks for the explanation, I will remove them and have assertions that you proposed in https://github.com/strongloop/loopback-next/pull/1974#discussion_r230689362.